chore(@xen-orchestra/backups): mutualize modules

badrAZ opened the pull request:
Test done:

- [x] test deduped
- [x] file restore on simple partition
- [x] delta backup with CR
- [x] full backup with DR
- [x] restore full VM backup
- [x] restore delta VM backup
- [x] XO and pool metadata backup
- [x] restore XO metadata backup
- [x] restore pool metadata backup

### Check list

> Check if done, if not relevant leave unchecked.

- [ ] PR reference the relevant issue (e.g. `Fixes #007` or `See xoa-support#42`)
- [ ] if UI changes, a screenshot has been added to the PR
- [ ] documentation updated
- `CHANGELOG.unreleased.md`:
  - [ ] enhancement/bug fix entry added
  - [ ] list of packages to release updated (`${name} v${new version}`)
- **I have tested added/updated features** (and impacted code)
  - [ ] unit tests (e.g. [`cron/parse.spec.js`](https://github.com/vatesfr/xen-orchestra/blob/b24400b21de1ebafa1099c56bac1de5c988d9202/%40xen-orchestra/cron/src/parse.spec.js))
  - [ ] if `xo-server` API changes, the corresponding test has been added to/updated on [`xo-server-test`](https://github.com/vatesfr/xen-orchestra/tree/master/packages/xo-server-test)
  - [ ] at least manual testing

### Process

1. create a PR as soon as possible
1. mark it as `WiP:` (Work in Progress) if not ready to be merged
1. when you want a review, add a reviewer (and only one)
1. if necessary, update your PR, and re- add a reviewer

From [_the Four Agreements_](https://en.wikipedia.org/wiki/Don_Miguel_Ruiz#The_Four_Agreements):

1. Be impeccable with your word.
1. Don't take anything personally.
1. Don't make assumptions.
1. Always do your best.

badrAZ committed:
chore(@xen-orchestra/backups): mutualize modules

badrAZ committed:
update implementation

badrAZ committed:
update implementation

badrAZ committed:
update implementation

julien-f commented on @xen-orchestra/backups/_ContinuousReplicationWriter.js:
```diff
@@ -116,3 +115,5 @@ export class ContinuousReplicationWriter {
   }
 }
+
+exports.ContinuousReplicationWriter = ContinuousReplicationWriter
```
Why not exporting it directly?

```js
exports.ContinuousReplicationWrite = class ContinuousReplicationWrite {
```

badrAZ replied:
I  did this type of export for all modules. And , this type of practice is done before in the library.

julien-f replied:
Less code, less repetition, closer to ES modules.

julien-f commented on @xen-orchestra/backups/_PoolMetadataBackup.js:
```diff
@@ -73,3 +71,6 @@ export class PoolMetadataBackup {
   }
 }
+
+exports.PATH_DB_DUMP = PATH_DB_DUMP
```
I think you should export a value as close as possible to it's definition.

badrAZ replied:
For what reason?

julien-f replied:
Better code locality, closer to ES modules.

julien-f committed:
closer to ES modules

1. exports near definitions
2. inline exports when possible
3. ensure functions/classes are not anonymous

julien-f committed:
fixes

badrAZ committed:
update implementation

julien-f committed:
chore(proxy): remove unused deps

badrAZ committed:
update implementation

badrAZ committed:
update implementation

badrAZ committed:
update implementation

badrAZ committed:
Merge remote-tracking branch 'origin/proxy-v1' into mutualizeBackupModules
